### ISSUE TITLE
[Bug]: Search result is empty for collection which has data

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#28338]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20231109-b29edeba
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka): kafka   
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: (api_request)  : [Collection] args: ['deploy_test_index_type_HNSW_is_compacted_is_compacted_segment_status_all_is_string_indexed_is_string_indexed_replica_number_1_is_deleted_is_deleted_data_size_3000', {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True......, kwargs: {'consistency_level': 'Strong'} (api_request.py:62)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: (api_response) : <Collection>:

[2023-11-09T08:37:10.869Z] -------------

[2023-11-09T08:37:10.869Z] <name>: deploy_test_index_type_HNSW_is_compacted_is_compacted_segment_status_all_is_string_indexed_is_string_indexed_replica_number_1_is_deleted_is_deleted_data_size_3000

[2023-11-09T08:37:10.869Z] <description>: 

[2023-11-09T08:37:10.869Z] <schema>: {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'descrip......  (api_request.py:37)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - INFO - ci_test]: inserted 3000 data into collection deploy_test_index_type_HNSW_is_compacted_is_compacted_segment_status_all_is_string_indexed_is_string_indexed_replica_number_1_is_deleted_is_deleted_data_size_3000 (common_func.py:1382)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: Dynamic field is enabled: False (common_func.py:1384)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                                       float_vector

[2023-11-09T08:37:10.869Z] 0         0     0.0       0  [0.10149285163328774, 0.09623293541853473, 0.1...

[2023-11-09T08:37:10.869Z] 1         1     1.0       1  [0.13323703873876852, 0.0520657931352484, 0.10...

[2023-11-09T08:37:10.869Z] 2         2     2.0       2  [0.12729870105101299, 0.064500200......, kwargs: {'timeout': 180} (api_request.py:62)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 445518686622842882, success count: 3000, err count: 0)  (api_request.py:37)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:15 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.03769262102369912, 0.12348943900031377, 0.12953062973687773, 0.085886262734884, 0.14460795020215705, 0.01023205261802302, 0.060215505307852975, 0.024446208152313492, 0.05325974907662942, 0.1030413620240372, 0.02363245567785696, 0.12627170710374502, 0.14238024009467118, 0.054613366444885424, 0.0......, kwargs: {} (api_request.py:62)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:22 - DEBUG - ci_test]: (api_response) : []  (api_request.py:37)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:22 - INFO - ci_test]: search_results_check: checking the searching results (func_check.py:287)

[2023-11-09T08:37:10.869Z] [2023-11-09 08:35:22 - ERROR - ci_test]: search_results_check: Numbers of query searched (0) is not equal with expected (2) (func_check.py:303)[get_env_variable] failed to get environment variables : 'CI_LOG_PATH', use default path : /tmp/ci_logs
```

collection name: deploy_test_index_type_HNSW_is_compacted_is_compacted_segment_status_all_is_string_indexed_is_string_indexed_replica_number_1_is_deleted_is_deleted_data_size_3000

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_kafka_for_release_cron/detail/deploy_test_kafka_for_release_cron/1533/pipeline

log:
[artifacts-kafka-standalone-reinstall-1533-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/13316178/artifacts-kafka-standalone-reinstall-1533-pytest-logs.tar.gz)
[artifacts-kafka-standalone-reinstall-1533-server-second-deployment-logs.tar.gz](https://github.com/milvus-io/milvus/files/13316180/artifacts-kafka-standalone-reinstall-1533-server-second-deployment-logs.tar.gz)

**Comment by zhuwenxing** [comment]
It seems like a stable issue.

failed job: 
* https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_kafka_for_release_cron/detail/deploy_test_kafka_for_release_cron/1533/pipeline
* https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_kafka_for_release_cron/detail/deploy_test_kafka_for_release_cron/1538/pipeline

**Comment by zhuwenxing** [comment]
Verified and fixed in 2.2.0-20231110-4e60e6ac
